build.gradle

```diff
@@ -33,6 +33,9 @@ subprojects {
         implementation group: 'com.google.inject', name: 'guice', version: guiceVersion
         // https://mvnrepository.com/artifact/commons-cli/commons-cli
         implementation group: 'commons-cli', name: 'commons-cli', version: commonsCliVersion
+        // https://search.maven.org/artifact/tools.aqua/z3-turnkey/4.8.16/jar
+        implementation 'tools.aqua:z3-turnkey:4.8.16'
```

Comment:
I noticed that this added dependency is not reflected in the OSGi or Maven configurations. This means that it would only work with `lfc`, which is built using Gradle, not with Epoch or the LDS server (and thus VS Code).

Reply:
For Maven there is this:
```
<dependency>
    <groupId>tools.aqua</groupId>
    <artifactId>z3-turnkey</artifactId>
    <version>4.8.16</version>
</dependency>
```
I don't know how to fix the Eclipse configuration, to be honest. But we can't merge this until that is fixed because otherwise the project simply won't build in Eclipse.

Reply:
Maybe @a-sr has any suggestions?

Reply:
Add this, but Maven still doesn't work. I'm probably missing something obvious again...

Reply:
Alternatively, z3 could be included as a submodule. I expect this kind of problem to surface with other dependencies, too. And including a submodule for every library that isn't available as an OSGi bundle doesn't seem like a good solution either.

Reply:
Yes, all non-plugin dependencies in Eclipse are tricky. If z3 is a compile-time dependency for Java, I guess there is no way around putting it in a plugin to make it work in Eclipse.
We also had this issue in Klighd with the Sprotty library that is only available on maven central. We created a repository to wrap it into a plugin ([sprotty-mirror](https://github.com/kieler/sprotty-mirror)). The build automatically pulls the library, builds it into a plugin, and deploys it as an update site, such that we can use it in our Eclipse and Oomph setup as dependency. Maybe we need something similar for LF.
The official Eclipse Orbrit works with the same principle.

Regarding the dependency you added to the pom file, I do not know why this does not work or if that even will work.  In the maven configuration, the option to consider pom dependencies in addition to the target platform is enables but I do not know how it works in detail. The current configuration has something to do with the workaround for Kotlin, but to be honest I am not sure how all these configurations work together.
Sorry.